features/admin.sms-templates.v1/pages/sms-customization.tsx

```diff
@@ -0,0 +1,457 @@
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { Dispatch } from "redux";
+import { Divider, Segment } from "semantic-ui-react";
+import {
+    createNewSmsTemplate,
+    deleteSmsTemplate,
+    updateSmsTemplate,
+    useSmsTemplate,
+    useSmsTemplatesList
+} from "../api/sms-templates";
+import SmsCustomizationFooter from "../components/sms-customization-footer";
+import { SmsCustomizationForm } from "../components/sms-customization-form";
+import SmsCustomizationHeader from "../components/sms-customization-header";
+import { SmsTemplatePreview } from "../components/sms-template-preview";
+import { SmsTemplate, SmsTemplateType } from "../models/sms-templates";
+import "./sms-customization.scss";
+
+type SmsCustomizationPageInterface = IdentifiableComponentInterface;
+
+/**
+ * SMS customization page.
+ *
+ * @param props - Props injected to the component.
+ *
+ * @returns Main Page for SMS Customization.
+ */
+const SmsCustomizationPage: FunctionComponent<SmsCustomizationPageInterface> = (
+    props: SmsCustomizationPageInterface
+): ReactElement => {
+
+    const { ["data-componentid"]: componentId } = props;
+
+    const [ availableSmsTemplatesList, setAvailableSmsTemplatesList ] = useState<SmsTemplateType[]>([]);
+    const [ currentSmsTemplate, setCurrentSmsTemplate ] = useState<SmsTemplate>();
+    const [ isSystemTemplate, setIsSystemTemplate ] = useState(true);
+    const [ isTemplateNotAvailable, setIsTemplateNotAvailable ] = useState(false);
+    const [ selectedLocale, setSelectedLocale ] = useState(I18nConstants.DEFAULT_FALLBACK_LANGUAGE);
+    const [ selectedSmsTemplateId, setSelectedSmsTemplateId ] = useState<string>();
+    const [ selectedSmsTemplateDescription, setSelectedSmsTemplateDescription ] = useState<string>();
+    const [ selectedSmsTemplate, setSelectedSmsTemplate ] = useState<SmsTemplate>();
+
+    const smsTemplates: Record<string, string>[] = useSelector(
+        (state: AppState) => state.config.deployment.extensions.smsTemplates) as Record<string, string>[];
+    const enableCustomSmsTemplates: boolean = useSelector(
+        (state: AppState) => state?.config?.ui?.enableCustomSmsTemplates);
+    const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
+    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
+    const smsFeatureConfig: FeatureAccessConfigInterface = useSelector(
+        (state: AppState) => state.config.ui.features.smsTemplates);
+
+    const dispatch: Dispatch = useDispatch();
+    const { t } = useTranslation();
+    const { getLink } = useDocumentation();
+    const hasUpdatePermission : boolean = useRequiredScopes(smsFeatureConfig?.scopes?.update);
+    const hasCreatePermission : boolean = useRequiredScopes(smsFeatureConfig?.scopes?.create);
+
+    const isReadOnly: boolean = useMemo(() => {
```

Comment:
No need to use useMemo for this. Next one as well. Get rid of them and define a constant and use it.